### PR TITLE
XWIKI-18910: Errors appear in the wiki console after selecting the "Forgot your username or password option", on a Multilingual wiki

### PR DESCRIPTION
**Pull request by surli**
JIRA: https://jira.xwiki.org/browse/XWIKI-18910

Proposed solution:
  * Ensure to display the language menu only if $doc is defined in the
    current context.

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/layoutvars.vm [hunk 1]
```diff
@@ -119,6 +119,9 @@
  #end
 #end
 
+## Check that the doc velocity variable is defined: it might not be the case when using some resource handlers.
+#set ($docIsDefined = ("$!doc" != ''))
```
Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/layoutvars.vm:123]: Not sure why you compare it to an empty String, a simple `#if ($doc)` (which check if it's not null) would be enough IMO. Feels more accurate than calling XWikiDocument#toString().

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/layoutvars.vm [hunk 2]
```diff
@@ -131,7 +134,9 @@
 #set($displayShortcuts = true)
 #set($displayPageHeader = false) ## page header containing the logo and the banner taken from ColorTheme
 #set ($displayQuickSearch = true)
-#set ($displayLanguageChoice = true)
+
+## the language choice only makes sense when a doc is declared.
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/layoutvars.vm:138]: Minor :)
```suggestion
## The language choice only makes sense when a doc is declared.
```

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/layoutvars.vm (hunk 1 above)
Comment (surli) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/layoutvars.vm:123]: I'm using this pattern when I want a one-liner, it avoids
```
#if ($doc)
  #set ($docIsDefined = true)
#else 
  #set ($docIsDefined = false)
#end 
```
but if you have a better way for that kind of stuff I'm taking it.

Comment (surli) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/layoutvars.vm:123]: hmm maybe I could use `$doc != $NULL`?

Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/layoutvars.vm:123]: That should work, another possibility is `#set ($docIsDefined = !!$doc)` (putting a `!` in front of a variable in a set cause the set to set as value the boolean result of the if and then you invert it again with another `!`).

Comment (surli) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/layoutvars.vm:123]: Hmm I prefer `$doc != $NULL` it's easier to read IMO.

**Review by manuelleduc — APPROVED** [review]

**Pull request merged**